src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx

```diff
@@ -39,13 +74,78 @@ export const PosterPreview: React.FC<PosterPreviewProps> = ({
     [categories]
   );
 
+  const { relatedVenueIds } = useRelatedVenues({
+    //isLoading, relatedVenues,
+    currentVenueId: venueId,
+  });
+
+  const {
+    // isEventsLoading,
+    events: relatedVenueEvents = emptyRelatedEvents,
+  } = useVenueEvents({
+    venueIds: relatedVenueIds,
+  });
+
+  const bookmarkPoster = useCallback(() => {
+    setBookmarkPoster(!isBookmarkedPoster);
+    personalizedPoster.isSaved = !personalizedPoster.isSaved;
+    if (userId && personalizedPoster.id) {
+      savePosterToProfile({
+        venueId: personalizedPoster.id,
+        userId: userId,
+        removeMode: !personalizedPoster.isSaved,
+      });
+    }
+    relatedVenueEvents
+      .filter((event) => event.venueId === personalizedPoster.id)
+      .map((event) => {
+        userId &&
+          event.id &&
+          updatePersonalizedSchedule({
+            event: event,
+            userId: userId,
+            removeMode: !personalizedPoster.isSaved,
+          });
+        return {};
+      });
+    return relatedVenueEvents;
+  }, [userId, isBookmarkedPoster, personalizedPoster, relatedVenueEvents]);
+
+  const onBookmarkPoster: MouseEventHandler<HTMLDivElement> = useCallback(
+    (e) => {
+      e.stopPropagation();
+      bookmarkPoster();
+    },
+    [bookmarkPoster]
+  );
+
+  // wrap author in memo:
+  // const { worldUsersById } = useWorldUsersById();
+  // const authors = personalizedPoster.owners;
+  // const author = { ...worldUsersById[authors[0]], id: authors[0] };
+  // const { openUserProfileModal } = useProfileModalControls();
+  // const openAuthorProfile = useCallback(() => {
+  //   openUserProfileModal(author);
+  // }, [openUserProfileModal, author]);
+
   return (
     <div className={posterClassnames} onClick={handleEnterVenue}>
+      <div className="PosterPreview__bookmark" onClick={onBookmarkPoster}>
+        <FontAwesomeIcon
+          icon={isBookmarkedPoster ? solidBookmark : regularBookmark}
+        />
```

Comment:
Since we've used this 2 times now, and I can imagine it making sense to be used elsewhere in future as well, we should extract this into a tiny atom component, that takes a single `boolean` prop. Then use it both here and in `src/components/molecules/ScheduleEvent/ScheduleEvent.tsx` (and anywhere else that uses this sort of icon, if anywhere else exists) currently.

Reply:
I've extracted this now to an atom component